dronegen/push.go

```diff
@@ -2,40 +2,64 @@ package main
 
 import "fmt"
 
-type buildType struct {
+type pushBuildType struct {
 	os   string
 	arch string
 	fips bool
 }
 
-// makefileTarget gets the correct Makefile target for a given arch/fips combo
-func makefileTarget(params buildType) string {
+// pushMakefileTarget gets the correct push pipeline Makefile target for a given arch/fips combo
+func pushMakefileTarget(params pushBuildType) string {
 	makefileTarget := fmt.Sprintf("release-%s", params.arch)
 	if params.fips {
 		makefileTarget += "-fips"
 	}
 	return makefileTarget
 }
 
+// pushCheckoutCommands builds a list of commands for Drone to check out a git commit on a push build
+func pushCheckoutCommands(fips bool) []string {
+	commands := []string{
+		`mkdir -p /go/src/github.com/gravitational/teleport /go/cache`,
+		`cd /go/src/github.com/gravitational/teleport`,
+		`git init && git remote add origin ${DRONE_REMOTE_URL}`,
+		`git fetch origin`,
+		`git checkout -qf ${DRONE_COMMIT_SHA}`,
+		// this is allowed to fail because pre-4.3 Teleport versions don't use the webassets submodule
+		`git submodule update --init webassets || true`,
+		`mkdir -m 0700 /root/.ssh && echo "$GITHUB_PRIVATE_KEY" > /root/.ssh/id_rsa && chmod 600 /root/.ssh/id_rsa`,
+		`ssh-keyscan -H github.com > /root/.ssh/known_hosts 2>/dev/null && chmod 600 /root/.ssh/known_hosts`,
+		`git submodule update --init e`,
+		// do a recursive submodule checkout to get both webassets and webassets/e
+		// this is allowed to fail because pre-4.3 Teleport versions don't use the webassets submodule
+		`git submodule update --init --recursive webassets || true`,
+		`rm -f /root/.ssh/id_rsa`,
+	}
+	if fips {
+		commands = append(commands, `if [[ "${DRONE_TAG}" != "" ]]; then echo "${DRONE_TAG##v}" > /go/.version.txt; else egrep ^VERSION Makefile | cut -d= -f2 > /go/.version.txt; fi; cat /go/.version.txt`)
```

Comment:
nit: any reason this won't use `make print-version` to extract the version?

Reply:
Embarrassingly, I hadn't even noticed that was a thing. I think originally I was doing everything to avoid installing `make` inside every single Docker image, but that does seem quite a lot simpler to use.